hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/impl/ChangeTracker.java

```diff
@@ -148,16 +177,77 @@ public void processResponse(final S3Object object,
       }
     }
 
-    final ObjectMetadata metadata = object.getObjectMetadata();
+    processMetadata(object.getObjectMetadata(), operation, pos);
+  }
+
+  /**
+   * Process the response from the server for validation against the
+   * change policy.
+   * @param copyResult result of a copy operation
+   * @throws PathIOException raised on failure
+   * @throws RemoteFileChangedException if the remote file has changed.
+   */
+  public void processResponse(final CopyResult copyResult)
```

Comment:
Maybe I shouldn't have even put this method in here given it doesn't do anything (at least right now).  I added it before I realized the ETag and versionId wouldn't (necessarily) be the same on the copied object.  With certain encryption algorithms, ETag will actually be the same and I suppose we could try to compare in those cases but it felt too awkward to bother with.

I'm curious for feedback.  Should I remove this method altogether?  Or leave it as a means to document that it wasn't just forgotten?

Reply:
Actually, one thing I can do here is enforce fs.s3a.change.detection.version.required (if set) to make sure the CopyResult has an ETag or versionId if one is expected.  I'll add that.

Reply:
+1, and maybe add debug logging as well

Reply:
Updated